contracts/src/v0.8/ccip/pools/TokenPool.sol

```diff
@@ -224,16 +235,55 @@ abstract contract TokenPool is IPoolV1, Ownable2StepMsgSender {
     return s_remoteChainConfigs[remoteChainSelector].remoteTokenAddress;
   }
 
-  /// @notice Sets the remote pool address for a given chain selector.
-  /// @param remoteChainSelector The remote chain selector for which the remote pool address is being set.
-  /// @param remotePoolAddress The address of the remote pool.
-  function setRemotePool(uint64 remoteChainSelector, bytes calldata remotePoolAddress) external onlyOwner {
+  /// @notice Adds a remote pool for a given chain selector. This could be due to a pool being upgraded on the remote
+  /// chain. We don't simply want to replace the old pool as there could still be valid inflight messages from the old
+  /// pool. This function allows for multiple pools to be added for a single chain selector.
+  /// @param remoteChainSelector The remote chain selector for which the remote pool address is being added.
+  /// @param remotePoolAddress The address of the new remote pool.
+  function addRemotePool(uint64 remoteChainSelector, bytes calldata remotePoolAddress) external onlyOwner {
     if (!isSupportedChain(remoteChainSelector)) revert NonExistentChain(remoteChainSelector);
 
-    bytes memory prevAddress = s_remoteChainConfigs[remoteChainSelector].remotePoolAddress;
-    s_remoteChainConfigs[remoteChainSelector].remotePoolAddress = remotePoolAddress;
+    if (keccak256(remotePoolAddress) == EMPTY_ENCODED_ADDRESS_HASH || remotePoolAddress.length == 0) {
```

Comment:
hm this is an evm specific check since EMPTY_ENCODED_ADDRESS_HASH is abi.encoded. Although its unlikely that 32 byte 0x0 is a valid remote pool on some non-EVM, it could be arbitrary bytes so you in principle you shouldn't assume any structure. Slight preference for leaving it out just for cross family consistency